From 4b58fbd93bcd214b975851de3f284066b173d58c Mon Sep 17 00:00:00 2001
From: nxf24178 <ravindra.yashvant.shinde@nxp.com>
Date: Mon, 14 Feb 2022 15:47:46 +0530
Subject: [PATCH] [artf934823]: Remove third party module from system partition

The Vendor specific framework code has been moved to product partition,
a part of framework which is tightly coupled with AOSP framework has been
moved to system_ext partition.
---
 Android.bp                                                 | 7 +++++--
 src/com/android/nfc/NfcService.java                        | 2 +-
 src/com/android/nfc/cardemulation/RegisteredAidCache.java  | 2 +-
 .../android/nfc/cardemulation/RegisteredServicesCache.java | 2 +-
 4 files changed, 8 insertions(+), 5 deletions(-)

diff --git a/Android.bp b/Android.bp
index e6d04d3d..c12c0945 100644
--- a/Android.bp
+++ b/Android.bp
@@ -19,11 +19,14 @@ android_app {
         "nci/**/*.java",
         ":statslog-Nfc-java-gen",
     ],
-    uses_libs: ["com.nxp.nfc"],
+    uses_libs: ["com.nxp.nfc",
+                "com.nxp.nfcintf",
+               ],
     platform_apis: true,
     certificate: "platform",
     jni_libs: ["libnfc_nci_jni"],
-    libs: ["com.nxp.nfc"],
+    libs: ["com.nxp.nfc",
+           "com.nxp.nfcintf"],
     static_libs: [
         "androidx.appcompat_appcompat",
         "android.hardware.secure_element-V1.0-java",
diff --git a/src/com/android/nfc/NfcService.java b/src/com/android/nfc/NfcService.java
index f4474060..7f96bba2 100644
--- a/src/com/android/nfc/NfcService.java
+++ b/src/com/android/nfc/NfcService.java
@@ -122,7 +122,7 @@ import com.nxp.nfc.INxpNfcAdapter;
 import com.nxp.nfc.INxpNfcAdapterExtras;
 import com.nxp.nfc.INxpWlcAdapter;
 import com.nxp.nfc.INxpWlcCallBack;
-import com.nxp.nfc.NfcConstants;
+import com.nxp.nfcintf.NfcConstants;
 import com.nxp.nfc.NxpNfcAdapter;
 
 import java.io.BufferedReader;
diff --git a/src/com/android/nfc/cardemulation/RegisteredAidCache.java b/src/com/android/nfc/cardemulation/RegisteredAidCache.java
index 81e9e483..a84da1b4 100644
--- a/src/com/android/nfc/cardemulation/RegisteredAidCache.java
+++ b/src/com/android/nfc/cardemulation/RegisteredAidCache.java
@@ -61,7 +61,7 @@ import java.util.Map;
 import java.util.NavigableMap;
 import java.util.PriorityQueue;
 import java.util.TreeMap;
-import com.nxp.nfc.NfcConstants;
+import com.nxp.nfcintf.NfcConstants;
 
 public class RegisteredAidCache {
     static final String TAG = "RegisteredAidCache";
diff --git a/src/com/android/nfc/cardemulation/RegisteredServicesCache.java b/src/com/android/nfc/cardemulation/RegisteredServicesCache.java
index c4b56609..18266207 100644
--- a/src/com/android/nfc/cardemulation/RegisteredServicesCache.java
+++ b/src/com/android/nfc/cardemulation/RegisteredServicesCache.java
@@ -58,7 +58,7 @@ import android.util.Log;
 import android.util.SparseArray;
 import android.util.Xml;
 import android.util.proto.ProtoOutputStream;
-import com.nxp.nfc.NfcConstants;
+import com.nxp.nfcintf.NfcConstants;
 
 import com.android.internal.annotations.GuardedBy;
 import com.android.internal.util.FastXmlSerializer;
-- 
2.35.1

